packages/daf-web-did/src/identity-controller.ts

```diff
@@ -0,0 +1,55 @@
+import { AbstractIdentityController, AbstractKeyManagementSystem, AbstractIdentityStore } from 'daf-core'
+import Debug from 'debug'
+const debug = Debug('daf:web-did:identity-controller')
+
+export class IdentityController extends AbstractIdentityController {
+  private did: string
+  private kms: AbstractKeyManagementSystem
+  private identityStore: AbstractIdentityStore
+
+  constructor(options: {
+    did: string
+    kms: AbstractKeyManagementSystem
+    identityStore: AbstractIdentityStore
+  }) {
+    super()
+    this.did = options.did
+    this.kms = options.kms
+    this.identityStore = options.identityStore
+  }
+
+  async addService(service: { id: string; type: string; serviceEndpoint: string }): Promise<any> {
+    return Promise.reject('[daf-web-did identity-controller] addService not implemented')
```

Comment:
If it isn't already, I think the identity store should be able to deal with extra fields in the serialized identity.
Isn't it possible to use that to store service blocks as well?

Reply:
I created an issue for this: https://github.com/uport-project/daf/issues/172